sklearn/metrics/_pairwise_distances_reduction/__init__.py

```diff
@@ -46,56 +46,56 @@
 #
 #
 #                               (base dispatcher)
-#                           PairwiseDistancesReduction
+#                                   Dispatcher
 #                                       ∆
 #                                       |
 #                                       |
-#                     +-----------------+-----------------+
-#                     |                                   |
-#               (dispatcher)                        (dispatcher)
-#         PairwiseDistancesArgKmin           PairwiseDistancesRadiusNeighbors
+#               +-----------------------+----------------------+
 #               |                                              |
+#          (dispatcher)                                   (dispatcher)
+#       ArgKminDispatcher                        RadiusNeighborhoodDispatcher
```

Comment:
I think we could make those dispatcher names even shorter:

```suggestion
#               |                                              |
#          (dispatcher)                                   (dispatcher)
#            ArgKmin                                   RadiusNeighborhood
```

Those are the classes that will be used in other modules of scikit-learn. The calling code doesn't care that it's calling the actual computation through a dispatching intermediate class.

The fact that those classes are dispatching classes is already reflected in the name of their base class.